pkg/builder/kaniko/kaniko.go

```diff
@@ -0,0 +1,248 @@
+package kaniko
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/celestiaorg/knuu/pkg/builder"
+	"github.com/google/uuid"
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	kanikoImage         = "gcr.io/kaniko-project/executor:debug" // debug has a shell
+	kanikoContainerName = "kaniko-container"
+	kanikoJobNamePrefix = "kaniko-build-job"
+)
+
+type Kaniko struct {
+	k8sClientset kubernetes.Interface
+	k8sNamespace string
+}
+
+var _ builder.Builder = &Kaniko{}
+
+func NewKaniko(k8sClientset kubernetes.Interface, namespace string) *Kaniko {
```

Comment:
because we're already inside kaniko package, I'd rename this New, so it reads `kaniko.New` vs `kaniko.NewKaniko`

potentially would be more idiomatic to pass in an options struct and have it just take one argument with the k8sClientset and namespace in the options IMO

Reply:
delicate point, thanks. will apply it asap